src/js/components/TextArea/TextArea.js

```diff
@@ -1,14 +1,42 @@
 import React, { Component } from 'react';
 import { compose } from 'recompose';
 
+import { Keyboard } from '../Keyboard';
 import { withFocus, withForwardRef, withTheme } from '../hocs';
 
 import { StyledTextArea } from './StyledTextArea';
 
 class TextArea extends Component {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { forwardRef } = nextProps;
+    const { inputRef } = prevState;
+    const nextInputRef = forwardRef || inputRef;
+    if (nextInputRef !== inputRef) {
+      return { inputRef: nextInputRef };
+    }
+    return null;
+  }
+
+  state = {
+    inputRef: React.createRef(),
+  };
+
+  onEsc = event => {
+    const { inputRef } = this.state;
+    if (event.keyCode === 27 && document.activeElement === inputRef.current) {
+      event.nativeEvent.stopImmediatePropagation(); // so Layer doesn't close
+      inputRef.current.blur();
+    }
+  };
+
   render() {
     const { fill, forwardRef, ...rest } = this.props;
-    return <StyledTextArea ref={forwardRef} fillArg={fill} {...rest} />;
+    const { inputRef } = this.state;
```

Comment:
why we have to put the ref in the state? I saw this in the drop button and removed it. it presented problems with performance too.